bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/FloureonThermostatHandler.java

```diff
@@ -0,0 +1,254 @@
+                            SensorControl.INTERNAL_TEMP_EXTERNAL_LIMIT);
+                }
+            } catch (Exception e) {
+                logger.error("Error while trying to set sensor mode {}: ", command, e);
+            }
+        } else {
+            logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    private void handleRemoteLockCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof OnOffType) {
+            try {
+                floureonDevice.setLock(command == OnOffType.ON);
+            } catch (Exception e) {
+                logger.error("Error while setting remote lock of {} to {}", thing.getUID(), command, e);
+            }
+        } else {
+            logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    private void handleSetTimeCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof DateTimeType) {
+            ZonedDateTime zonedDateTime = ((DateTimeType) command).getZonedDateTime();
+            try {
+                new SetTimeCommand(tob(zonedDateTime.getHour()), tob(zonedDateTime.getMinute()),
+                        tob(zonedDateTime.getSecond()), tob(zonedDateTime.getDayOfWeek().getValue()))
+                                .execute(floureonDevice);
+            } catch (Exception e) {
+                logger.error("Error while setting time of {} to {}", thing.getUID(), command, e);
+            }
+        } else {
+            logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    @Override
+    protected void refreshData() {
+        if (ThingStatus.ONLINE != thing.getStatus()) {
+            return;
+        }
+        try {
+            AdvancedStatusInfo advancedStatusInfo = floureonDevice.getAdvancedStatus();
+            if (advancedStatusInfo == null) {
+                logger.warn("Device {} did not return any data. Trying to reauthenticate...", thing.getUID());
+                authenticate();
+                advancedStatusInfo = floureonDevice.getAdvancedStatus();
+            }
+            if (advancedStatusInfo == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Device not responding.");
+                return;
+            }
+            logger.debug("Retrieved data from device {}: {}", thing.getUID(), advancedStatusInfo);
+            logger.debug("Updating channel {} with value {}", ROOM_TEMPERATURE,
+                    new DecimalType(advancedStatusInfo.getRoomTemp()));
+            logger.debug("Mode {}", StringType.valueOf(advancedStatusInfo.getAutoMode() ? "auto" : "manual"));
+            updateState(ROOM_TEMPERATURE, new DecimalType(advancedStatusInfo.getRoomTemp()));
+            logger.debug("Updating channel {} with value {}", ROOM_TEMPERATURE_EXTERNAL_SENSOR,
+                    new DecimalType(advancedStatusInfo.getExternalTemp()));
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
I kept the very first one as this was indeed helpful for debugging. The remaining ones are removed.